tensornetwork/tensornetwork_test.py

```diff
@@ -791,6 +791,33 @@ def build_tensornetwork(tensors):
     result = tf.map_fn(build_tensornetwork, tensors, dtype=tf.float32)
     self.assertAllClose(result, tf.ones(5) * 10)
 
+  def test_weakref(self):
+    net = tensornetwork.TensorNetwork()
+    a = net.add_node(np.eye(2))
+    b = net.add_node(np.eye(2))
+    e = net.connect(a[0], b[0])
+    del a
+    del b
+    net.contract(e)
+    with self.assertRaises(ValueError):
+      e.node1
+
+  def test_weakref_complex(self):
+    net = tensornetwork.TensorNetwork()
+    a = net.add_node(np.eye(2))
+    b = net.add_node(np.eye(2))
+    c = net.add_node(np.eye(2))
+    e1 = net.connect(a[0], b[0])
+    e2 = net.connect(b[1], c[0])
+    net.contract(e1)
+    net.contract(e2)
+    # This won't raise an exception since we still have a referance to 'a'.
+    e1.node1
+    with self.assertRaises(ValueError):
+      e2.node1
```

Comment:
Could use a comment about why this raises an exception.

Reply:
Done.